libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/ApolloResponse.kt

```diff
@@ -132,10 +132,18 @@ private constructor(
       this.extensions = extensions
     }
 
+    fun operation(operation: Operation<D>) = apply {
+      this.operation = operation
+    }
```

Comment:
I don't think it makes sense to allow changing the `operation`. Once you have a `ApolloResponse<D>`, the `D` type parameter is bound to the operation so this cannot work:

```kotlin
val response = ApolloResponse<MyQuery.Data>(myQuery, uuid, data)

val myMytation = MyMutation()

// Expecting Operation<MyQuery.Data>, not MyMutation.Data
response.newBuilder().operation(myMutation)
```

Reply:
Agreed 👍